### PR TITLE
Add size constants and properties for grid areas

### PR DESCRIPTION
**Pull request by NCrashed**
Hi, I hit missing properties for sizes and grid areas and added them into my project. So, that it is draft for possible additions. I am not sure if experimental CSS API is acceptable in clay. Also, more consistent part of grid areas API should be added to the PR.

Use case:
``` CSS
.grid1 {
  display: grid;
  width: max-content;
}

.grid3 {
  display: grid;
  width: max-content;
}

@media (min-width: 40.0rem) {
  .grid3 {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-gap: 1rem;
    width: max-content;
  }
}

```
``` haskell
  ".grid1" ? do
    display grid
    width maxContent
  ".grid3" ? do
    display grid
    width maxContent
  query M.screen [M.minWidth (rem 40)] $ ".grid3" ? do
    display grid
    gridTemplateColumns [fr 1, fr 1, fr 1]
    gridGap $ rem 1
    width maxContent
```

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -6,5 +6,6 @@ in release.pkgs.stdenv.lib.overrideDerivation release.clay.env (oldAttrs: rec {
   nativeBuildInputs = (oldAttrs.nativeBuildInputs or []) ++ [
     release.cabal
     release.pkgs.haskellPackages.cabal2nix
+    release.pkgs.haskellPackages.ghcid
```
Comment (turion) [inline shell.nix]: Why is this needed?

File: src/Clay/Size.hs [hunk 2]
```diff
@@ -165,6 +170,25 @@ vmin i = SimpleSize (cssDoubleText i <> "vmin")
 -- | SimpleSize in vmax's (the larger of vw or vh).
 vmax i = SimpleSize (cssDoubleText i <> "vmax")
 
+-- | SimpleSize in fr's (a fractional unit and 1fr is for 1 part of the available space in grid areas).
```
Comment (turion) [inline src/Clay/Size.hs]:
```suggestion
-- | 'SimpleSize' in fr's (a fractional unit and 1fr is for 1 part of the available space in grid areas).
```
Actually I'd say it's a good idea using haddock links wherever possible.

File: src/Clay/Size.hs [hunk 3]
```diff
@@ -165,6 +170,25 @@ vmin i = SimpleSize (cssDoubleText i <> "vmin")
...
+
+-- | SimpleSize for the intrinsic preferred width.
+maxContent :: Size LengthUnit
+maxContent = SimpleSize "max-content"
+
+-- | SimpleSize for the intrinsic minimum width.
+minContent :: Size LengthUnit
+minContent = SimpleSize "min-content"
+
+-- | SimpleSize for The containing block width minus horizontal margin, border, and padding.
```
Comment (turion) [inline src/Clay/Size.hs]:
```suggestion
-- | SimpleSize for the containing block width minus horizontal margin, border, and padding.
```

File: src/Clay/Size.hs [hunk 4]
```diff
@@ -165,6 +170,25 @@ vmin i = SimpleSize (cssDoubleText i <> "vmin")
...
+available :: Size LengthUnit
+available = SimpleSize "available"
+
+-- | The larger of the intrinsic minimum width or the smaller of the intrinsic preferred width and the available width
```
Comment (turion) [inline src/Clay/Size.hs]:
```suggestion
-- | The larger of the intrinsic minimum width or the smaller of the intrinsic preferred width and the available width.
```

**Comment by turion** [comment]
I think it's a good idea to add your usage example to the module docs of your new `Clay.Grid` module.

File: shell.nix (hunk 1 above)
Comment (NCrashed) [reply shell.nix]: Oops, it is left from my toolchain. It is allows to watch after changes in ghci and automatically reload changed modules.

**Comment by NCrashed** [comment]
> Thanks for your contribution! I don't understand the CSS aspects well enough to judge this, so ideally you could ping someone who does, for a short review. (If you don't know anyone, no worries.)
> 
> Your PR is marked as a draft. What's missing?

Unfortunately, I don't know such experts to call them :). I marked the PR as draft as I feel it doesn't cover all related API for grid areas. So, if it is enough to be merged lets cover the rest of API in subsequent PRs.

File: shell.nix (hunk 1 above)
Comment (turion) [reply shell.nix]: No worries :) it should be removed from the PR though.

**Comment by turion** [comment]
> Unfortunately, I don't know such experts to call them :)

No worries. I think this PR is good, we can merge it. If you want you can open separate issues that explain what should be added to the API to complete it.

**Comment by turion** [comment]
Thanks!
